strider/query_planner.py

```diff
@@ -225,58 +291,61 @@ async def qg_to_og(qgraph):
             # Forward inverse edge
             ograph["edges"][edge_id + INVERSE_EDGE_SUFFIX] = make_og_edge(
                 edge_id, edge,
-                edge_reverse=True,
-                predicate_reverse=False,
+                edge_reverse=False,
+                predicate_reverse=True,
                 predicates=inverse_predicates)
             # Reverse inverse edge
             ograph["edges"][edge_id + REVERSE_EDGE_SUFFIX + INVERSE_EDGE_SUFFIX] = \
                 make_og_edge(
                 edge_id, edge,
-                edge_reverse=False,
-                predicate_reverse=True,
+                edge_reverse=True,
+                predicate_reverse=False,
                 predicates=inverse_predicates
             )
 
     return ograph
 
 
-def filter_categories_predicates(operation_graph):
+def fix_categories_predicates(query_graph):
     """
-    Filter out categories and predicates that are not defined in any KP
+    Given a permuted query graph with one KP,
+    fix the node categories and predicates to match the KP
+
+    Returns false if there is a conflicting node type
```

Comment:
It seems probably more "pythonic" to raise an exception for conflicting categories rather than returning a boolean.